app/controllers/activitypub/inboxes_controller.rb

```diff
@@ -52,6 +53,20 @@ def upgrade_account
     DeliveryFailureTracker.reset!(signed_request_account.inbox_url)
   end
 
+  def process_collection_synchronization
+    raw_params = request.headers['X-AS-Collection-Synchronization']
```

Comment:
Don't want to bikeshed but my suggestion would be to save the few bytes and just name it `Collection-Synchronization`. It does not conflict with any standard headers.